core/src/main/scala/io/delta/tables/DeltaColumnBuilder.scala

```diff
@@ -126,6 +127,10 @@ class DeltaColumnBuilder private[tables](
       metadataBuilder.putString("comment", comment.get)
     }
     val fieldMetadata = metadataBuilder.build()
+    if (dataType == null) {
+      throw DeltaErrors.analysisException("dataType should not be null for a column," +
```

Comment:
nit: `s"The date type of column $colName is not provided"`

Reply:
Thanks, this new error message has the right amount of information 👍